core/src/main/java/com/google/instrumentation/trace/PropagationUtil.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.trace;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.common.io.BaseEncoding;
+import java.util.concurrent.atomic.AtomicReferenceArray;
+
+/**
+ * This is a helper class for {@link SpanContext} propagation on the wire. Allows changing the
+ * defautl handler at runtime using {@link #setHandler(Handler)}.
```

Comment:
Removed the entire sentence because for the moment setHandler is package-protected.